File: tests/07_phases_ompss.dg/fortran/success_task_24.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>
PROGRAM P
    IMPLICIT NONE
    INTEGER, POINTER :: PTR(:, :)

    ALLOCATE(PTR(10, 5))

    PTR = 1

    !$OMP TARGET DEVICE(SMP) COPY_DEPS
    !$OMP TASK INOUT(PTR)
        PTR(1, :)= -2
    !$OMP END TASK

    !$OMP TASKWAIT
    if (ANY(PTR(1,:) /= -2)) STOP 1
    if (ANY(PTR(2:,:) /= 1)) STOP 2
END PROGRAM P

File: tests/07_phases_ompss.dg/c/success_task_24.c
/*
<testinfo>
test_generator=config/mercurium-ompss
</testinfo>
*/
#include <assert.h>

void compute(const int N, const int TS, double (*a)[TS][TS])
{
    int i;
    for (i = 0; i < N; i++)
    {
        #pragma omp task inout(a[i][0;TS][0;TS])
        {
            int j, k;
            for (j = 0; j < TS; ++j)
                for (k = 0; k < TS; ++k)
                    a[i][j][k]++;
        }
    }
}

void init(const int N, const int TS, double (*a)[TS][TS])
{
    int i;
    for (i = 0; i < N; i++)
    {
        #pragma omp task out(a[i][0;TS][0;TS])
        {
            int j, k;
            for (j = 0; j < TS; ++j)
                for (k = 0; k < TS; ++k)
                    a[i][j][k] = 0.0;
        }
    }
}

void check(const int N, const int TS, double (*a)[TS][TS])
{
    int i;
    for (i = 0; i < N; i++)
    {
        #pragma omp task in(a[i][0;TS][0;TS])
        {
            int j, k;
            for (j = 0; j < TS; ++j)
                for (k = 0; k < TS; ++k)
                    assert(a[i][j][k] == 1.0);
        }
    }
}

int main(int argc, char* argv[])
{
    const int N = 10;
    const int TS = 25;

    double v[N][TS][TS];

    init(N, TS, v);
    compute(N, TS, v);
    check(N, TS, v);

    #pragma omp taskwait
    return 0;
}
